### PR TITLE
[ADD] [15.0] stock_warn_option

### PR DESCRIPTION
**Pull request by Shide**
This module extends the functionality of blocking messages to support more consistency
between messages and to allow you to set a blocking message quickly.

This is a PR of a series:

[base_warn_option](https://github.com/OCA/server-ux/pull/853)
[sale_warn_option](https://github.com/OCA/sale-workflow/pull/3076)
[purchase_warn_option](https://github.com/OCA/purchase-workflow/pull/2240)
[account_warn_option](https://github.com/OCA/account-invoicing/pull/1704)

MT-5193 @moduon @rafaelbn @yajo @EmilioPascual please review if you want :)

### CODE REVIEW

File: stock_warn_option/tests/test_warn_options.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    def test_partner_warn_options(self):
+        """Test Warn Options on Partner Form"""
+        with Form(self.partner) as partner_f:
+            partner_f.picking_warn = "warning"
+            partner_f.picking_warn_option = self.partner_picking_warn_warning
+            self.assertEqual(partner_f.picking_warn_msg, "warning")
+            partner_f.picking_warn = "block"
+            partner_f.picking_warn_option = self.partner_picking_warn_blocking
+            self.assertEqual(partner_f.picking_warn_msg, "block")
+        partner_f.save()
```
Comment (rousseldenis) [inline stock_warn_option/tests/test_warn_options.py]: This one is not needed as `__exit__()` in Form will do it.

**Review by rousseldenis — APPROVED** [review]
Code review. Minor comment

File: stock_warn_option/tests/test_warn_options.py (hunk 1 above)
Comment (Shide) [reply stock_warn_option/tests/test_warn_options.py]: Done :)

**Review by yajo — APPROVED** [review]

**Pull request merged**